### PR TITLE
Fix local memory reference & logging

### PR DESCRIPTION
**Pull request by guusw**
Fix for something that might have caused a crash in the infos.clj test

### CODE REVIEW

File: src/core/shards/casting.hpp [hunk 1]
```diff
@@ -249,7 +249,7 @@ template <SHType ToType> struct MakeVector {
     static Parameters params = []() {
       Parameters params;
       auto vectorType = VectorTypeLookup::getInstance().get(ToType);
-      auto paramType = getCompatibleUnitType(vectorType->numberType);
+      const Type &paramType = getCompatibleUnitType(vectorType->numberType);
```
Comment (sinkingsugar) [inline src/core/shards/casting.hpp:252]: ouch

**Review by sinkingsugar — APPROVED** [review]

File: src/core/shards/casting.hpp (hunk 1 above)
Comment (Kryptos-FR) [reply src/core/shards/casting.hpp:252]: Just curious: what wouldn't `auto` grab the correct type here? Or should have it been `auto&`?

**Review by Kryptos-FR — APPROVED** [review]

File: src/core/shards/casting.hpp (hunk 1 above)
Comment (guusw) [reply src/core/shards/casting.hpp:252]: auto always tries to remove reference types, so you explicitly need to use auto& when you want to store a reference.
(doesn't remove pointers)

Comment (guusw) [reply src/core/shards/casting.hpp:252]: I think const carries over, so you can assign a const reference to `auto& something = ...`

**Pull request merged**